### ISSUE TITLE
ak.unflatten doesn't seem to work for axis>0

### ISSUE THREAD

**Issue by Dominic-Stafford** [scikit-hep/awkward#819]
Using `ak.unflatten` with `axis=1` gives an error for awkward v1.2.1:
```
>>> a = ak.Array([[1,2,3,4],[5,6,7,8]])
>>> ak.unflatten(a, 2, axis=0)
<Array [[[1, 2, 3, 4], [5, 6, 7, 8]]] type='1 * 2 * var * int64'>
>>> ak.unflatten(a, 2, axis=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/afs/desy.de/user/s/stafford/.local/lib/python3.8/site-packages/awkward/operations/structure.py", line 2091, in unflatten
    out = ak._util.recursively_apply(
  File "/afs/desy.de/user/s/stafford/.local/lib/python3.8/site-packages/awkward/_util.py", line 1302, in recursively_apply
    return apply(layout, 1, user)
  File "/afs/desy.de/user/s/stafford/.local/lib/python3.8/site-packages/awkward/_util.py", line 1098, in apply
    custom = getfunction(layout, *args)
  File "/afs/desy.de/user/s/stafford/.local/lib/python3.8/site-packages/awkward/operations/structure.py", line 2059, in getfunction
    inneroffsets = nplike.asarray(content.offsets)
AttributeError: 'awkward._ext.RegularArray' object has no attribute 'offsets'
```
Using an array of depth 3 I got the error for both  axis=1 and axis=2.

Thank you

Dominic

**Comment by jpivarski** [comment]
Yeah,

```python
>>> ak.unflatten(a, [2, 2, 2, 2], axis=1)
<Array [[[1, 2], [3, 4]], [[5, 6], [7, 8]]] type='2 * var * var * int64'>
```

works but the broadcasting in

```python
>>> ak.unflatten(a, 2, axis=1)
```

is failing.

**Comment by jpivarski** [comment]
Fixed in PR #820.

**Comment by Dominic-Stafford** [comment]
Thanks for fixing this so quickly!
